lib/active_model/serializer/associations.rb

```diff
@@ -84,17 +84,22 @@ def associate(reflection)
       # @return [Enumerator<Association>]
       #
       def associations(include_directive = ActiveModelSerializers.default_include_directive)
-        return unless object
-
         Enumerator.new do |y|
-          self.class._reflections.values.each do |reflection|
-            next if reflection.excluded?(self)
-            key = reflection.options.fetch(:key, reflection.name)
-            next unless include_directive.key?(key)
-            y.yield reflection.build_association(self, instance_options)
+          self.class._reflections.keys.each do |association_key|
+            assoc = association(association_key, include_directive)
+            next if assoc.nil?
+            y.yield assoc
           end
         end
       end
+
+      def association(association_key, include_directive = ActiveModelSerializers.default_include_directive)
+        return nil if object.nil?
+        reflection = _reflections[association_key]
+        return nil if reflection.excluded?(self)
+        return nil unless include_directive.key?(association_key)
```

Comment:
does an association include the resource identifier?

is the include directive the one that determines `included`ness? 
otherwise, how does the include_directive get populated?

Like, if I have an author, and the blogs relationship is rendered as just resource identifiers. How does this code play in to that?